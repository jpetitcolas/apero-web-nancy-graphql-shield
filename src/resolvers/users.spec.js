const { gql } = require('apollo-server-core');
const { print } = require('graphql');
const superagent = require('superagent');

describe('User Resolvers', () => {
    describe('getUsers', () => {
        const query = print(gql`
            query getUsers {
                users {
                    id
                    firstName
                    lastName
                    email
                }
            }
        `);

        it('should return a list of users', async () => {
            const response = await superagent.post('http://localhost:5000/graphql')
                .send({ query, variables: {} });

            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
    });
});
