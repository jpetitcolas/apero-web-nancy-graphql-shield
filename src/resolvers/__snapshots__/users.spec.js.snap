// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Resolvers getUsers should return a list of users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "email": "john.doe@example.com",
        "firstName": "John",
        "id": "1",
        "lastName": "Doe",
      },
      Object {
        "email": "alice.kellerman@example.com",
        "firstName": "Alice",
        "id": "2",
        "lastName": "Kellerman",
      },
    ],
  },
}
`;
